chart-diagrams/Graphics/Rendering/Chart/Backend/Diagrams.hs

```diff
@@ -126,7 +129,9 @@ cBackendToFile fo cb path = do
     EPS -> do
       let (d, a) = runBackend env cb
           opts = DEPS.PostscriptOptions path (D2.dims2D w h) DEPS.EPS
-      D.renderDia DEPS.Postscript opts d
+          eps = D.renderDia DEPS.Postscript opts d
+      withFile (opts D.^. DEPS.psfileName) WriteMode $ \h ->
```

Comment:
Why is this addition required?

Reply:
I was expecting the SMP changes (which look fine - thanks!)... but am unsure why this was also required...

Reply:
Ahh right. This is for https://github.com/diagrams/diagrams-postscript/commit/d64f48e9f7cf154799ada1fe056231c599d9e337 ... as diagrams-postscript SMP changes come after that commit. I guess haskell-chart is now dependent on a diagrams-postscript 1.5 release?

Reply:
Yep - that looks like a good solution!

Reply:
To me this change seems unrelated to making the package compile with GHC-8.4. By just adding `instance Semigroup Path where { (<>) = append }` the code in master compiled with `cabal install` and GHC-8.4.2.

Would it be possible to commit just the SMP changes and release a new version on Hackage?

Reply:
I'll try to split this into two tomorrow.
One that is SMP only, and affects only `chart` and another one that makes `chart-diagrams` SMP ready with a fix for the postscript 1.5 as detailed above.